File: MajorityElement.java
class Solution {
    public int majorityElement(int[] nums) {
        int res=0;
        int n= nums.length;
        
        for(int i:nums)
        {
            int count=0;
            for(int j :nums)
            {
                if(i==j)
                    count++;
                
            }
            if(count>n/2)
                    return i;
        }
        return -1;
    }
}

File: majorityElement.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n=nums.size();
        int can=0;
        int count=0;
        for(auto i:nums)
        {
            if(count==0)
            {
                can=i;
            }
            if(can==i)
            {
                count++;
            }
            else
            {
                count--;
            }
        }
        return can;
    }
};
